libs/partners/openai/tests/unit_tests/test_secrets.py

```diff
@@ -60,3 +65,336 @@ def test_azure_openai_embeddings_secrets() -> None:
     s = str(o)
     assert "foo1" not in s
     assert "foo2" not in s
+
+
+# azure chat openai
+def test_azure_chat_openai_api_key_is_secret_string() -> None:
+    """Test that the API key is stored as a SecretStr."""
+    chat_model = AzureChatOpenAI(
+        openai_api_key="secret-api-key",
+        azure_endpoint="endpoint",
+        azure_ad_token="secret-ad-token",
+        api_version="version",
+    )
+    assert isinstance(chat_model.openai_api_key, SecretStr)
+    assert isinstance(chat_model.azure_ad_token, SecretStr)
+
+
+def test_azure_chat_openai_api_key_masked_when_passed_from_env(
+    monkeypatch: MonkeyPatch, capsys: CaptureFixture
+) -> None:
+    """Test that the API key is masked when passed from an environment variable."""
+    monkeypatch.setenv("AZURE_OPENAI_API_KEY", "secret-api-key")
+    monkeypatch.setenv("AZURE_OPENAI_AD_TOKEN", "secret-ad-token")
+    chat_model = AzureChatOpenAI(
+        openai_api_key="secret-api-key",
```

Comment:
Should we remove the secrets from the constructor in this case, to enforce that they're read from the env vars?

Reply:
Yes. I was a tad too aggressive with the copy-paste.